roles/satellite-clone/tasks/interface_check.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Check if DNS is enabled
+  shell: tar zxf {{ backup_dir }}/config_files.tar.gz etc/foreman-installer/scenarios.d/satellite-answers.yaml --to-stdout | grep -m1 dns | awk '{print $2}'
+  register: dns_enabled
+
+- name: Check if DHCP is enabled
+  shell: tar zxf {{ backup_dir }}/config_files.tar.gz etc/foreman-installer/scenarios.d/satellite-answers.yaml --to-stdout | grep -m1 dhcp | awk '{print $2}'
+  register: dhcp_enabled
+
+- name: Grab DNS interface from Satellite answer file
+  shell: tar zxf {{ backup_dir }}/config_files.tar.gz etc/foreman-installer/scenarios.d/satellite-answers.yaml --to-stdout | grep dns_interface | awk '{print $2}'
+  register: dns_interface
+  when: dns_enabled.stdout == "true"
+
+- name: Grab DHCP interface from Satellite answer file
+  shell: tar zxf {{ backup_dir }}/config_files.tar.gz etc/foreman-installer/scenarios.d/satellite-answers.yaml --to-stdout | grep dhcp_interface | awk '{print $2}'
+  register: dhcp_interface
+  when: dhcp_enabled.stdout == "true"
+
+- name: Grab network interfaces from system
+  debug:
+    msg: "{{ ansible_interfaces | reject('match', 'lo') | list }}"
+  register: network_interfaces
+
+- name: Fail if interfaces on machine do not match answer file interfaces
+  fail:
+    msg: "The network interface {{ dns_interface.stdout }} setup for DNS and {{ dhcp_interface.stdout }} setup for DHCP do not match the network interfaces on the current machine"
+  when:
+    - dns_enabled.stdout == "true"
+    - dhcp_enabled.stdout == "true"
+    - dns_interface.stdout not in network_interfaces.msg
+    - dhcp_interface.stdout not in network_interfaces.msg
+    - check_networking_interfaces
```

Comment:
I thought I commented this, but maybe didn't submit the review! -- Since you are checking `check_networking_interfaces` to run this whole file, no need to check here.

Reply:
+1